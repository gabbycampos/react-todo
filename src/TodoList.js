import React, { useState } from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';

const TodoList = () => {
    const [todos, setTodos] = useState([]);

    const addTodo = (todo) => {
        setTodos((todos) => [...todos, todo]);
    };
    const remove = (id) => {
        setTodos((todos) => todos.filter((todo) => todo.id !== id));
    };

    const todoComponents = todos.map((todo) => (
        <Todo 
            key={todo.id}
            id={todo.id}
            name={todo.name}
            handleRemove={remove}
        />
    ));
    return (
        <div>
            <NewTodoForm addTodo={addTodo} />
            <ul>
            {todoComponents}
            </ul>
        </div>
    );
}

export default TodoList;