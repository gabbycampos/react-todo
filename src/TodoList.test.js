import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todoList, item = "buy coffee") {
  const taskInput = todoList.getByLabelText("Todo");
  fireEvent.change(taskInput, { target: { value: item } });
  const submitButton = todoList.getByText("Add Todo");
  fireEvent.click(submitButton);
}

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a todo", function () {
  const list = render(<TodoList />);
  addTodo(list);

  expect(list.getByLabelText("Todo")).toHaveValue("");
  expect(list.getByText("buy coffee")).toBeInTheDocument();

  expect(list.getByText("X")).toBeInTheDocument();
});

it("can delete a todo", function () {
  const list = render(<TodoList />);
  addTodo(list);

  fireEvent.click(list.getByText("X"));

  expect(list.queryByText("buy coffee")).not.toBeInTheDocument();
});