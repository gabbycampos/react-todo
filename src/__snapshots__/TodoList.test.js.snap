// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="nameForm"
        >
          Todo
        </label>
        <input
          id="nameForm"
          name="nameForm"
          type="text"
          value=""
        />
        <button>
          Add Todo
        </button>
      </form>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
